src/Sylius/Bundle/CoreBundle/Resources/config/services/listeners.xml

```diff
@@ -102,5 +102,11 @@
             <argument type="service" id="sylius.manager.order" />
             <tag name="kernel.event_listener" event="sylius.order.pre_complete" method="check" />
         </service>
+
+        <service id="sylius.listener.locker" class="Sylius\Bundle\CoreBundle\EventListener\LockerListener">
+            <argument type="service" id="doctrine.orm.entity_manager" />
+            <argument type="service" id="sylius.product_variant_resolver.default" />
+            <tag name="kernel.event_listener" event="sylius.product_variant.initialize" method="lock" />
```

Comment:
If we listen only for product variant events is it really possible to get a product?